lib/Smokeping.pm

```diff
@@ -1372,7 +1372,7 @@ sub get_detail ($$$$;$){
             my @lazy =();
             @lazy = ('--lazy') if $mode eq 's' and $lastheight{$s} and $lastheight{$s}{$start} and $lastheight{$s}{$start} == $max->{$s}{$start};
             my $timer_start = time();
-            my $from = $s ? " from $cfg->{Slaves}{$slave}{display_name}": "";
+            my $from = " from " . ($s ? $cfg->{Slaves}{$slave}{display_name}: $cfg->{General}{display_name} || hostname);
```

Comment:
I've taken the `||` syntax from the existing code base of SmokePing. And I don't like it either but I kept it for consistency. In the same file `SmokePing.pm` we [already have](https://github.com/famzah/SmokePing/blob/480dc9337dc2369054977132fdee97d2ebe8f4ff/lib/Smokeping.pm#L907):
```perl
if ($#slaves > 0){
    $label = sprintf("%-25s","median RTT from ".($slave ? $cfg->{Slaves}{$slave}{display_name} : $cfg->{General}{display_name} || hostname));
}
```

Regarding your second comment about the following:
```perl
$cfg->{General}{display_name} || hostname
```

If I understood the existing code and tested it properly, it's doing exactly what you say -- if there is no "display_name" configured in the "General" config, then use the hostname. According to the man page, "display_name" is optional in the "General" section.

---

I don't mind that we put parenthesis, in order to be more sure of the code correctness:
```perl
my $from = " from " . ($s ? $cfg->{Slaves}{$slave}{display_name}: ($cfg->{General}{display_name} || hostname) );
```